flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/BufferBuilderTestUtils.java

```diff
@@ -29,4 +29,10 @@ public static BufferBuilder createBufferBuilder(int size) {
 			MemorySegmentFactory.allocateUnpooledSegment(size),
 			FreeingBufferRecycler.INSTANCE);
 	}
+
+	public static Buffer buildSingleBuffer(BufferBuilder bufferBuilder) {
+		try (BufferConsumer bufferConsumer = bufferBuilder.createBufferConsumer()) {
+			return bufferConsumer.build();
+		}
```

Comment:
Just double checking if this is not a problem: after this block, `close()` is called on the `BufferConsumer`, which recycles the buffer, but at the same time this method is returning a slice that is backed by to original buffer that is now recycled? Or it just doesn't matter for the tests?

Reply:
This is ok, since sliced buffer shares the same reference counter as the parent `BufferConsumer`. To recycle memory segment one needs to close all buffer consumers used for it AND recycle/release all created buffers from those buffer consumers.

Added comment explaining this to `BufferConsumer#build`.